File: premies/internal/grpc/mobile_bank.py
from gen.mobile_bank import mobile_bank_pb2, mobile_bank_pb2_grpc
from internal.service import mobile_bank


class MobileBankServiceServicer(mobile_bank_pb2_grpc.MobileBankServiceServicer):
    def UploadMobileBankData(self, request, context):
        try:
            result = mobile_bank.mobile_bank_excel_upload(request.file_path)
            return mobile_bank_pb2.MobileBankUploadResponse(status=result)
        except Exception as e:
            return mobile_bank_pb2.MobileBankUploadResponse(status=f"Error: {str(e)}")

    def CleanMobileBankTable(self, request, context):
        try:
            result = mobile_bank.mobile_bank_clean_table()
            return mobile_bank_pb2.MobileBankCleanResponse(status=result)
        except Exception as e:
            return mobile_bank_pb2.MobileBankCleanResponse(status=f"Error: {str(e)}")

File: internal/controllers/automation/mobile_bank.go
package automation

import (
	"context"
	"premiesPortal/internal/app/grpc/gen/mobile_bank"
	"premiesPortal/internal/app/service/validators"
	"premiesPortal/internal/clients/automation_premies/grpc"
	"premiesPortal/internal/controllers"
	"premiesPortal/pkg/errs"

	"github.com/gin-gonic/gin"
	"google.golang.org/protobuf/types/known/emptypb"
)

func UploadMobileBankData(c *gin.Context) {
	isValid, month := validators.ValidateMonth(c)
	if !isValid {
		controllers.HandleError(c, errs.ErrInvalidMonth)
		return
	}

	isValid, year := validators.ValidateYear(c)
	if !isValid {
		controllers.HandleError(c, errs.ErrInvalidYear)
		return
	}

	filePath := c.GetString(UploadedAutomationFilePath)

	if filePath == "" {
		controllers.HandleError(c, errs.ErrInvalidFilePath)
		return
	}

	client := grpc.GetClient()

	resp, err := client.UploadMobileBankData(context.Background(), &mobile_bank.MobileBankUploadRequest{
		Month:    int32(month),
		Year:     int32(year),
		FilePath: filePath,
	})
	if err != nil {
		controllers.HandleError(c, err)
		return
	}

	c.JSON(resp.StatusCode, gin.H{
		"message": resp.Resp,
	})
}

func CleanMobileBankTable(c *gin.Context) {
	client := grpc.GetClient()

	resp, err := client.CleanMobileBankTable(context.Background(), &emptypb.Empty{})
	if err != nil {
		controllers.HandleError(c, err)
		return
	}

	c.JSON(resp.StatusCode, gin.H{
		"message": resp.Resp,
	})
}
